x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/base_version_diff/base_version_flyout.tsx

```diff
@@ -0,0 +1,140 @@
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { EuiButton, EuiCallOut, EuiSpacer, EuiToolTip } from '@elastic/eui';
+import type { PartialRuleDiff, RuleResponse } from '../../../../../../common/api/detection_engine';
+import { PerFieldRuleDiffTab } from '../per_field_rule_diff_tab';
+import { RuleDetailsFlyout, TabContentPadding } from '../rule_details_flyout';
+import * as ruleDetailsI18n from '../translations';
+import * as i18n from './translations';
+import { RuleDiffTab } from '../rule_diff_tab';
+import { BaseVersionDiffFlyoutSubheader } from './base_version_flyout_subheader';
+import { useRevertPrebuiltRule } from '../../../logic/prebuilt_rules/use_revert_prebuilt_rule';
+
+export const PREBUILT_RULE_BASE_VERSION_FLYOUT_ANCHOR = 'baseVersionPrebuiltRulePreview';
+
+interface PrebuiltRulesBaseVersionFlyoutComponentProps {
+  currentRule: RuleResponse;
+  baseRule: RuleResponse;
+  diff: PartialRuleDiff;
+  closeFlyout: () => void;
+  isReverting: boolean;
+  onRevert?: () => void;
+}
+
+const PrebuiltRulesBaseVersionFlyoutComponent = ({
+  currentRule,
+  baseRule,
+  diff,
+  closeFlyout,
+  isReverting,
+  onRevert,
+}: PrebuiltRulesBaseVersionFlyoutComponentProps) => {
+  const { mutateAsync: revertPrebuiltRule, isLoading } = useRevertPrebuiltRule();
+  const subHeader = useMemo(
+    () => <BaseVersionDiffFlyoutSubheader currentRule={currentRule} diff={diff} />,
+    [currentRule, diff]
+  );
+
+  const ruleActions = useMemo(() => {
+    return isReverting ? (
+      <EuiButton
+        onClick={async () => {
+          await revertPrebuiltRule({
+            id: currentRule.id,
+            version: currentRule.version,
+            revision: currentRule.revision,
+          });
+          closeFlyout();
+          if (onRevert) {
+            onRevert();
+          }
+        }}
+        isDisabled={isLoading}
+        fill
+        data-test-subj="revertPrebuiltRuleFromFlyoutButton"
+      >
+        {i18n.REVERT_BUTTON_LABEL}
+      </EuiButton>
+    ) : null;
+  }, [
+    closeFlyout,
+    currentRule.id,
+    currentRule.revision,
+    currentRule.version,
+    isLoading,
+    isReverting,
+    onRevert,
+    revertPrebuiltRule,
+  ]);
+
+  const extraTabs = useMemo(() => {
+    const headerCallout = isReverting ? (
+      <>
+        <EuiCallOut title={i18n.REVERTING_RULE_CALLOUT_TITLE} color="warning" iconType="warning">
+          <p>{i18n.REVERTING_RULE_CALLOUT_MESSAGE}</p>
+        </EuiCallOut>
+        <EuiSpacer size="l" />
+      </>
+    ) : null;
+
+    const updatesTab = {
+      id: 'updates',
+      name: (
+        <EuiToolTip position="top" content={i18n.BASE_VERSION_FLYOUT_UPDATES_TAB_TOOLTIP}>
+          <>{'Elastic rule diff overview'}</>
+        </EuiToolTip>
+      ),
+      content: (
+        <TabContentPadding>
+          <PerFieldRuleDiffTab
+            header={headerCallout}
+            ruleDiff={diff}
```

Comment:
This is the reason I ended up returning the `diff` field in the `GetPrebuiltRuleBaseVersionResponseBody` type from the new `/base_version` endpoint. We can pass it directly to the existing `<PerFieldRuleDiffTab />` component without any modification and have it work.